evm_arithmetization/src/cpu/kernel/tests/core/access_lists.rs

```diff
@@ -10,82 +11,213 @@ use crate::cpu::kernel::constants::global_metadata::GlobalMetadata::{
+        .generation_state
+        .get_addresses_access_list()
+        .expect("Couldn't retrieve access list");
+
+    let Some((pos_0, next_val_0, _)) = list.next() else {
+        return Err(anyhow::Error::msg("Couldn't get value"));
+    };
+    assert_eq!(pos_0, 0);
+    assert_eq!(next_val_0, U256::MAX);
+    let Some((pos_0, next_val_0, _)) = list.next() else {
+        return Err(anyhow::Error::msg("Couldn't get value"));
+    };
+    assert_eq!(pos_0, 0);
+    Ok(())
+}
+
+#[test]
+fn test_insert_address() -> Result<()> {
+    let init_label = KERNEL.global_labels["init_access_lists"];
+
+    // Test for address already in list.
+    let initial_stack = vec![0xdeadbeefu32.into()];
+    let mut interpreter = Interpreter::<F>::new_with_kernel(init_label, initial_stack);
+    interpreter.run()?;
+
+    let insert_accessed_addresses = KERNEL.global_labels["insert_accessed_addresses"];
+
+    let retaddr = 0xdeadbeefu32.into();
+    let mut rng = thread_rng();
+    let mut address: H160 = rng.gen();
+
+    assert!(address != H160::zero(), "Cosmic luck or bad RNG?");
+
+    interpreter.push(retaddr);
+    interpreter.push(U256::from(address.0.as_slice()));
+    interpreter.generation_state.registers.program_counter = insert_accessed_addresses;
+
+    interpreter.run()?;
+    assert_eq!(interpreter.stack(), &[U256::one()]);
+    assert_eq!(
+        interpreter
+            .generation_state
+            .memory
+            .get(MemoryAddress::new_bundle(U256::from(AccessedAddressesLen as usize)).unwrap()),
+        U256::from(Segment::AccessedAddresses as usize + 4)
+    );
+
+    Ok(())
+}
+
 #[test]
 fn test_insert_accessed_addresses() -> Result<()> {
+    let init_access_lists = KERNEL.global_labels["init_access_lists"];
+
+    // Test for address already in list.
+    let initial_stack = vec![0xdeadbeefu32.into()];
+    let mut interpreter = Interpreter::<F>::new_with_kernel(init_access_lists, initial_stack);
+    interpreter.run()?;
+
     let insert_accessed_addresses = KERNEL.global_labels["insert_accessed_addresses"];
 
     let retaddr = 0xdeadbeefu32.into();
     let mut rng = thread_rng();
-    let n = rng.gen_range(1..10);
-    let addresses = (0..n)
+    let n = 10;
+    let mut addresses = (0..n)
         .map(|_| rng.gen::<Address>())
         .collect::<HashSet<_>>()
         .into_iter()
         .collect::<Vec<Address>>();
-    let addr_in_list = addresses[rng.gen_range(0..n)];
+    // The addresses must be sorted.
+    addresses.sort();
     let addr_not_in_list = rng.gen::<Address>();
     assert!(
         !addresses.contains(&addr_not_in_list),
         "Cosmic luck or bad RNG?"
     );
 
-    // Test for address already in list.
-    let initial_stack = vec![retaddr, U256::from(addr_in_list.0.as_slice())];
-    let mut interpreter: Interpreter<F> =
-        Interpreter::new_with_kernel(insert_accessed_addresses, initial_stack);
+    let offset = Segment::AccessedAddresses as usize;
     for i in 0..n {
```

Comment:
Instead of manually setting the memory, we might as well test `insert_accessed_addresses` here? It would make for a more robust test than `test_insert_address`.